src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Windows/Markup/ValueSerializer.cs

```diff
@@ -244,6 +244,9 @@ public static ValueSerializer GetSerializerFor(PropertyDescriptor descriptor, IV
         /// </summary>
         protected Exception GetConvertToException(object value, Type destinationType)
         {
+            if (destinationType == null)
```

Comment:
This one would already throw a NullReferenceException destinationType.FullName. By adding an explicit throw here, we make sure to throw early, with the right stacktrace and the right parameter name.

**This one is technically a breaking change, the type of exception returned is not the same as before. Let me know if you would prefer to revert this change.**

Reply:
Whilst this is a breaking change indeed, these types of changes have been generally considered acceptable across the .NET ecosystem.
https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/breaking-changes.md#bucket-2-reasonable-grey-area